### PR TITLE
AURORA: Add a text input callback to the gui class

### PR DESCRIPTION
**Pull request by Nostritius**
Adds a simple solution for gui wide text input over a callback. For deleting characters i thought about another callback for keys which could also be used for ingame shortcuts.

### CODE REVIEW

File: src/engines/aurora/gui.h [hunk 1]
```diff
@@ -114,6 +114,8 @@ class GUI : boost::noncopyable {
 	virtual void callbackRun();
 	/** Callback that's triggered when a widget was activated. */
 	virtual void callbackActive(Widget &widget);
+	/** Callback that's triggered when a text input is received */
```
Comment (DrMcCoy) [inline src/engines/aurora/gui.h]: The Doxygen comment should end with a period.

File: src/engines/aurora/gui.h [hunk 2]
```diff
@@ -114,6 +114,8 @@ class GUI : boost::noncopyable {
...
+	virtual void callbackTextInput(Common::UString text);
```
Comment (DrMcCoy) [inline src/engines/aurora/gui.h]: Again, this should probably be a const reference. The callback shouldn't usually need to modify the text input at all.

File: src/engines/aurora/gui.cpp [hunk 3]
```diff
@@ -508,6 +512,10 @@ void GUI::mouseWheel(const Events::Event &event) {
 	mouseWheel(_currentWidget, event);
 }
 
+void GUI::textInput(const Events::Event &event) {
```
Comment (DrMcCoy) [inline src/engines/aurora/gui.cpp]: P

File: src/engines/aurora/gui.cpp [hunk 4]
```diff
@@ -217,6 +219,8 @@ void GUI::callbackRun() {
 void GUI::callbackActive(Widget &UNUSED(widget)) {
 }
 
+void GUI::callbackTextInput(Common::UString text) {
```
Comment (DrMcCoy) [inline src/engines/aurora/gui.cpp]: Please wrap the `text` in the `UNUSED()` macro here, since the parameter isn't used in this method.

**Comment by DrMcCoy** [comment]
Again, except for a few style issues, this PR looks good to me. :)

**Comment by Nostritius** [comment]
Ok, this should now have corrected the style issues

**Comment by DrMcCoy** [comment]
Merged it as f9b464194, thanks! :)

**Pull request closed without merging**